Problem with oauth.d.ts: module has no default export

I guess this is more of a question than an actual issue, so bear with me.

With our `tsconfig.json` (see at the bottom of this issue), we get the following error: `ERROR in node_modules/ketting/dist/utils/oauth.d.ts(1,8): error TS1192: Module '"$WORKING_DIR/node_modules/client-oauth2/index"' has no default export.`.

The line in question is as follows: `import ClientOAuth2 from 'client-oauth2';`.  
When we change it to `import * as ClientOAuth2 from 'client-oauth2';`, everything works as expected.

Is this something in our `tsconfig.json`, is this something you can change, or is it something that's out of your hands?

Our `tsconfig.json`:

```
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "module": "es2015",
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2017",
      "dom"
    ]
  }
}
```

Thank you for your time (and this awesome library ;))

--- a/changelog.md
+++ b/changelog.md
@@ -6,6 +6,8 @@ ChangeLog
 
 * Basic JSON:API support. Currently only links appearing in the top-level
   document are supported.
+* Support for Typescript environments that don't have `esModuleInterop`
+  set to true.
 
 
 2.4.1 (2018-11-07)

--- a/changelog.md
+++ b/changelog.md
@@ -6,7 +6,7 @@ ChangeLog
 
 * Basic JSON:API support. Currently only links appearing in the top-level
   document are supported.
-* Support for Typescript environments that don't have `esModuleInterop`
+* #113: Support for Typescript environments that don't have `esModuleInterop`
   set to true.
 
 

--- a/src/link.ts
+++ b/src/link.ts
@@ -1,4 +1,4 @@
-import uriTemplate from 'uri-template';
+import * as uriTemplate from 'uri-template';
 import { resolve } from './utils/url';
 
 type LinkInit = {

--- a/src/representor/html.ts
+++ b/src/representor/html.ts
@@ -1,4 +1,4 @@
-import sax from 'sax';
+import * as sax from 'sax';
 import Link from '../link';
 import Representation from './base';
 

--- a/src/resource.ts
+++ b/src/resource.ts
@@ -1,4 +1,4 @@
-import LinkHeader from 'http-link-header';
+import * as LinkHeader from 'http-link-header';
 import FollowablePromise from './followable-promise';
 import problemFactory from './http-error';
 import Ketting from './ketting';

--- a/src/utils/oauth.ts
+++ b/src/utils/oauth.ts
@@ -1,4 +1,4 @@
-import ClientOAuth2 from 'client-oauth2';
+import * as ClientOAuth2 from 'client-oauth2';
 import { Token } from 'client-oauth2';
 import './fetch-polyfill';
 

--- a/src/utils/url.ts
+++ b/src/utils/url.ts
@@ -1,4 +1,4 @@
-import url from 'url';
+import { resolve as r } from 'url';
 
 /**
  * Resolves a relative url using another url.
@@ -7,7 +7,7 @@ import url from 'url';
  */
 export function resolve(base: string, relative: string): string {
 
-  return url.resolve(base, relative);
+  return r(base, relative);
 
 }
 

--- a/test/testserver.ts
+++ b/test/testserver.ts
@@ -1,11 +1,11 @@
-import fs from 'fs';
-import Koa from 'koa';
+import * as fs from 'fs';
+import * as Koa from 'koa';
 import { Context as KoaContext } from 'koa';
-import bodyParser from 'koa-bodyparser';
-import logger from 'koa-logger';
+import * as bodyParser from 'koa-bodyparser';
+import * as logger from 'koa-logger';
 // @ts-ignore don't have a definition for this.
-import Route from 'koa-path-match';
-import koaStatic from 'koa-static';
+import * as Route from 'koa-path-match';
+import * as koaStatic from 'koa-static';
 
 type Context = KoaContext & {
   params: { [s: string]: string }

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,7 +1,6 @@
 {
     "compilerOptions": {
         "module": "commonjs",
-        "esModuleInterop": true,
         "target": "es2017",
         "noImplicitAny": true,
 

